client/src/App.js

```diff
@@ -84,15 +79,16 @@ function App() {
     }, 7000);
   }
 
-  function set_LoggedUser(value) {
-    setLoggedUser(value);
-  }
+  useEffect(() => {
+    if (localStorage.loggedIn) {
+      set_new_user();
+    }
+  }, [triggerRerender]);
```

Comment:
This watches the `triggerRerender` for changes when that happens it will re-render.  Passing it through to the components in `onChange={setTriggerRerender}`